purldb-toolkit/src/purldb_toolkit/purlcli.py

```diff
@@ -0,0 +1,254 @@
+
+    if file:
+        purls = file.read().splitlines(False)
+
+    purl_versions = list_versions(purls)
+    json.dump(purl_versions, output, indent=4)
+
+
+def list_versions(purls):
+    purl_versions = []
+    for purl in purls:
+        dict_output = {}
+        dict_output["purl"] = purl
+        dict_output["versions"] = []
+
+        purl = purl.strip()
+        if not purl:
+            continue
+
+        if check_versions_purl(purl):
+            print(check_versions_purl(purl))
+            continue
+
+        for package_version_object in list(versions(purl)):
+            nested_dict = {}
+            nested_purl = purl + "@" + f'{package_version_object.to_dict()["value"]}'
+
+            nested_dict["purl"] = nested_purl
+            nested_dict["version"] = f'{package_version_object.to_dict()["value"]}'
+            nested_dict[
+                "release_date"
+            ] = f'{package_version_object.to_dict()["release_date"]}'
+
+            dict_output["versions"].append(nested_dict)
+
+        purl_versions.append(dict_output)
+
+    return purl_versions
+
+
+@purlcli.command(name="meta")
+@click.option(
+    "--purl",
+    "purls",
+    multiple=True,
+    required=False,
+    help="PackageURL or PURL.",
+)
+@click.option(
+    "--output",
+    type=click.File(mode="w", encoding="utf-8"),
+    required=True,
+    default="-",
+    help="Write versions output as JSON to FILE.",
+)
+@click.option(
+    "--file",
+    type=click.File(mode="r", encoding="utf-8"),
+    required=False,
+    help="Read a list of PURLs from a FILE, one per line.",
+)
+def get_meta(purls, output, file):
+    """
+    Given one or more PURLs, return a mapping of metadata fetched from the API for each PURL.
+    """
+    if purls and file:
+        raise click.UsageError("Use either purls or file but not both.")
+
+    if not (purls or file):
+        raise click.UsageError("Use either purls or file.")
+
+    if file:
+        purls = file.read().splitlines(False)
+
+    meta_info = get_meta_details(purls)
+    json.dump(meta_info, output, indent=4)
+
+
+def get_meta_details(purls):
+    meta_details = []
+    purl_versions = []
+
+    for purl in purls:
+        dict_output = {}
+        dict_output["purl"] = purl
+        dict_output["metadata"] = []
+
+        purl = purl.strip()
+        if not purl:
+            continue
+
+        if check_meta_purl(purl):
+            print(check_meta_purl(purl))
+            continue
+
+        releases = []
+        for release in list(info(purl)):
+            dict_output["metadata"].append(release.to_dict())
+
+        purl_versions.append(dict_output)
+
+    return purl_versions
+
+
+def check_existence(purl):
+    api_query = "https://public.purldb.io/api/validate/"
+    purl = purl.strip()
+    request_body = {"purl": purl, "check_existence": True}
+    response = requests.get(api_query, params=request_body)
+    results = response.json()
+
+    return results
+
+
+def check_versions_purl(purl):
```

Comment:
Add a docstring describing what the function does

Reply:
Docstring added.